File: apps/backend/app/schemas/product.py
import uuid
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict, Field

class ProductCreateRequest(BaseModel):
    product_name: str = Field(..., min_length=1, max_length=255)
    cost_price: Decimal = Field(..., ge=0)
    selling_price: Decimal = Field(..., ge=0)
    current_stock: int = Field(0, ge=0)
    low_stock_alert: int = Field(0, ge=0)

class ProductUpdateRequest(BaseModel):
    product_name: str | None = Field(None, min_length=1, max_length=255)
    cost_price: Decimal | None = Field(None, ge=0)
    selling_price: Decimal | None = Field(None, ge=0)
    current_stock: int | None = Field(None, ge=0)
    low_stock_alert: int | None = Field(None, ge=0)
    is_active: bool | None = None

class ProductResponse(BaseModel):
    id: uuid.UUID
    product_name: str
    cost_price: Decimal
    selling_price: Decimal
    current_stock: int
    low_stock_alert: int
    is_active: bool
    created_at: datetime
    updated_at: datetime
    model_config = ConfigDict(from_attributes=True)

File: apps/web/types/product.ts
export interface ProductResponse {
  id: string;
  merchant_id: string;
  product_name: string;
  cost_price: number;
  selling_price: number;
  current_stock: number;
  low_stock_alert?: number | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProductCreatePayload {
  product_name: string;
  cost_price: number;
  selling_price: number;
  current_stock: number;
  low_stock_alert?: number | null;
}

export interface ProductUpdatePayload {
  product_name?: string;
  cost_price?: number;
  selling_price?: number;
  current_stock?: number;
  low_stock_alert?: number | null;
  is_active?: boolean;
}
